Neos.ContentRepository.Core/Classes/Infrastructure/DbalCheckpointStorage.php

```diff
@@ -94,7 +94,7 @@ public function acquireLock(): SequenceNumber
             ]);
         } catch (DBALException $exception) {
             $this->connection->rollBack();
-            if ($exception instanceof LockWaitTimeoutException || ($exception instanceof DBALDriverException && ($exception->getErrorCode() === 3572 || $exception->getErrorCode() === 7))) {
+            if ($exception instanceof LockWaitTimeoutException) {
```

Comment:
Probably one of the more questionable changes I had to make, but the driver exception no longer carries an error code so we can't do this comparison anymore...

Reply:
🤷‍♂️

Reply:
According to the docs in `\Doctrine\DBAL\Driver\AbstractException`, the `getCode` seems to be the replacement we are looking for? so `$exception->getCode() === 3572`?

`getSQLState()`: The SQLSTATE the driver is in at the time the error occurred, if any.
`getCode()`: The driver specific error code if any.

Actually i found out at least why we check agains `3572`, it seems this is a special case and doctrine doesnt throw a `LockWaitTimeoutException` here see https://github.com/doctrine/dbal/issues/6048

Reply:
Mmmm, right, that could work, the question is do we want to or just throw the actual exception like it is now?

Reply:
by removing the check and throwing the original exception we would be reverting @bwaidelich 's change where the code originates from: https://github.com/neos/eventstore-doctrineadapter/commit/0f530010e34ed75082006be164fc9fa087f5db6a

so i dont know ... and also the https://github.com/doctrine/dbal/issues/6048 sounds legit... i just dont know where the `7` comes from

Reply:
7 was the corresponding code on PostgreSQL IIRC..
But that will all be obsolete when switching to the lock based catchup

Reply:
right, well your decision @bwaidelich , we can quickly change it back to use getCode() and hope it works as expected or ignore it for now with the plan that it gets removed anyways?

Reply:
@kitsunet I'm sorry, I can only do these half-reviews atm – is the change related to the DBAL update at all?

Reply:
Yeah, the `getErrorCode` was removed and as mhs linked above (which I didn'T see) `getCode()` can be used as replacement

Reply:
uppdated to use getCode()